pkg/tempo/internal/tempoutils/server.go

```diff
@@ -73,121 +73,144 @@ func NewServerWithRandomPort(callback func(pdata.Traces)) (srv *Server, addr str
 // NewServer creates an OTLP-accepting server that calls a function when a
 // trace is received. This is primarily useful for testing.
 func NewServer(addr string, callback func(pdata.Traces)) (*Server, error) {
-	conf := util.Untab(fmt.Sprintf(`
-processors:
-	func_processor:
-receivers:
-  otlp:
-		protocols:
-			grpc:
-				endpoint: %s
-service:
-	pipelines:
-		traces:
-			receivers: [otlp]
-			processors: [func_processor]
-			exporters: []
-	`, addr))
+	factories, err := tracingFactories(callback)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating tracing factories: %s", err)
+	}
+
+	var (
+		startInfo component.ApplicationStartInfo
+	)
 
-	var cfg map[string]interface{}
-	if err := yaml.NewDecoder(strings.NewReader(conf)).Decode(&cfg); err != nil {
-		panic("could not decode config: " + err.Error())
+	params := service.Parameters{
+		Factories:            factories,
+		ApplicationStartInfo: startInfo,
```

Comment:
Nitpick: this can be simplified a tiny bit, avoiding a var block.

```suggestion
		ApplicationStartInfo: component.ApplicationStartInfo{},
```

Reply:
Applied, thanks